### PR TITLE
feat(query): Added Encryption Provider Not Properly Configured for Kubernetes

### PR DESCRIPTION
**Pull request by cxMiguelSilva**
**Proposed Changes**
- Added Encryption Provider Not Properly Configured for Kubernetes

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/k8s/encryption_provider_not_properly_configured/metadata.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "id": "10efce34-5af6-4d83-b414-9e096d5a06a9",
+  "queryName": "Encryption Provider Not Properly Configured",
+  "severity": "MEDIUM",
+  "category": "Encryption",
+  "descriptionText": "The EncryptionConfiguration should configured to have at least one 'aescbc' provider",
```
Comment (joaorufi) [inline assets/queries/k8s/encryption_provider_not_properly_configured/metadata.json:6]:
```suggestion
  "descriptionText": "The EncryptionConfiguration should be configured to have at least one 'aescbc' provider",
```

File: assets/queries/k8s/encryption_provider_not_properly_configured/query.rego [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package Cx
+
+import data.generic.common as common_lib
+import data.generic.k8s as k8sLib
+
+CxPolicy[result] {
+	resource := input.document[i]
+	resource.kind == "EncryptionConfiguration"
+	not conatainsProvider(resource, "aescbc")
```
Comment (joaorufi) [inline assets/queries/k8s/encryption_provider_not_properly_configured/query.rego:9]:
```suggestion
	not containsProvider(resource, "aescbc")
```

File: assets/queries/k8s/encryption_provider_not_properly_configured/query.rego [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": "kind={{EncryptionConfiguration}}.providers",
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": "aescbc provider should be defined",
+		"keyActualValue": "aescbc provider is not defined",
+	}
+}
+
+conatainsProvider(resource, providerToCheck){
```
Comment (joaorufi) [inline assets/queries/k8s/encryption_provider_not_properly_configured/query.rego:20]:
```suggestion
containsProvider(resource, providerToCheck){
```

**Review by joaorufi — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/encryption_provider_not_properly_configured/query.rego [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package Cx
+
+import data.generic.common as common_lib
+import data.generic.k8s as k8sLib
+
+CxPolicy[result] {
+	resource := input.document[i]
+	resource.kind == "EncryptionConfiguration"
+	not containsProvider(resource, "aescbc")
```
Comment (rafaela-soares) [inline assets/queries/k8s/encryption_provider_not_properly_configured/query.rego:9]: Maybe we should also look for kms or secretbox

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rafaela-soares — DISMISSED** [review]
LGTM 🚀

**Review by joaorufi — APPROVED** [review]

**Pull request merged**